### PR TITLE
fix(commandbar): fix popup not dismissing when a pointer clicks outside it

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): #13147

## PR Type

What kind of change does this PR introduce?
- Bugfix

### <samp>🤖 Generated by Copilot at 7143b80</samp>

Improve popups for `CommandBar` on iOS and macOS. Fix closing and hit testing issues in `PopupPanel.cs`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs [hunk 1]
```diff
@@ -245,8 +245,8 @@ private void OnPointerPressed(object sender, PointerRoutedEventArgs args)
 		)
 		{
 			// The check is here because ContentDialogPopupPanel returns true for IsViewHit() even though light-dismiss is always
-			// disabled for ContentDialogs
-			if (popup.IsLightDismissEnabled)
+			// disabled for ContentDialogs. In addition, CommandBar is always closes the Popup.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:248]: Probably the popup should have had IsLightDismissEnabled in the first place

Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:248]: Nope. That's the problem. It dismisses regardless of IsLightDismissEnabled.

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:248]: Interesting. Could you try setting `IsSticky` on Windows and see if it changes the behavior?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:248]: I think your change should have failed `ValidateInlineCommandBarLightDismissBehavior`:

https://github.com/unoplatform/uno/blob/e5b91a50ed1e3d469b24f4e67caad1b0c7c8741f/src/Uno.UI.RuntimeTests/IntegrationTests/dxaml/controls/commandbar/CommandBarIntegrationTests.cs#L988

I don't see why this test didn't fail, but that means we have a problem that the test doesn't validate the behavior properly (somehow).

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:248]: The dismiss should happen through `TryDismissInlineAppBar` (likely in `OnOverlayElementPointerPressed`).

See https://github.com/unoplatform/uno/blob/e5b91a50ed1e3d469b24f4e67caad1b0c7c8741f/src/Uno.UI/UI/Xaml/Controls/AppBar/AppBar.Partial.cs#L1305


and

https://github.com/unoplatform/uno/blob/e5b91a50ed1e3d469b24f4e67caad1b0c7c8741f/src/Uno.UI/UI/Xaml/Controls/AppBar/AppBar.Partial.cs#L1281-L1291

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs [hunk 2]
```diff
@@ -241,12 +241,19 @@ private protected override void OnUnloaded()
...
+			// in AppBar.SetupOverlayState) but we don't have the logic implemented in Uno yet, so we
+			// rely on this workaround to close CommandBar's popup.
+			if (popup.TemplatedParent is CommandBar cb)
+			{
+				cb.TryDismissInlineAppBarInternal();
+			}
 			// The check is here because ContentDialogPopupPanel returns true for IsViewHit() even though light-dismiss is always
-			// disabled for ContentDialogs
-			if (popup.IsLightDismissEnabled)
+			// disabled for ContentDialogs. In addition, CommandBar is always closes the Popup.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:255]: > In addition, CommandBar is always closes the Popup.

I think this part of the comment can be removed.

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs [hunk 3]
```diff
@@ -241,12 +241,19 @@ private protected override void OnUnloaded()
...
 		// Make sure we are the original source.  We do not want to handle PointerPressed on the Popup itself.
-		if (args.OriginalSource == this && Popup is { } popup
-		)
+		if (args.OriginalSource == this && Popup is { } popup)
 		{
+			// CommandBars in WinUI don't rely on IsLightDismissEnabled, instead there's IsSticky.
+			// Instead of handling it here, CommandBar should handle it using an LTE (look at the comment
+			// in AppBar.SetupOverlayState) but we don't have the logic implemented in Uno yet, so we
+			// rely on this workaround to close CommandBar's popup.
+			if (popup.TemplatedParent is CommandBar cb)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:250]: Looks good, are you able to make a UI test for it? Runtime may be tricky.

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:250]: Not a big a fan of specific cases like this which are tightly coupling unrelated classes ... but acceptable as a **temp** workaround :/

**Pull request merged**